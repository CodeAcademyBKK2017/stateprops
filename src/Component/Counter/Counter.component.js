import React from 'react';
import PropTypes from 'prop-types';

const Counter = (props) => {
  const myThis = props;
  return <h1 style={myThis.style}>{myThis.count}</h1>;
};

Counter.propTypes = {
  text: PropTypes.number.isRequired
};

Counter.defaultProps = {
  text: 0
};

export default Counter;
