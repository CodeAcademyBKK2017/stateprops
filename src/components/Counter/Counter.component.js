import React, {Component} from 'react';
import './Counter.style.css';

class Counter extends Component {
	render () {
		const {text} = this.props;

		return (
			<div className="Counter">
				{text}
			</div>
		);
	}
}

export default Counter;
