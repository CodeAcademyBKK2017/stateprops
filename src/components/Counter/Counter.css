.box-result {
    margin: 100px 0px;
}

.counter {
    width: 240px;
    height: 180px;

    margin: 0 auto;

    font-size: 48px;
    font-weight: bold;
    color: black;
    border: 3px solid black;
    background-color: #33b5ff;
    text-align: center;
    line-height: 180px;
}