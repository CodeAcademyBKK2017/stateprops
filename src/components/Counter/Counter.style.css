.Counter-box {
    display: inline-block;
    border: 3px solid black;
    background-color: #27b4ff;
    color: black;
    font-size: 60px;
    width: 220px;
    height: 220px;
    text-align: center;
}
.Counter-button {
    display: inline-block;
    border: 3px solid black;
    border-radius: 10px;
}
.Counter-button--yellow {
    background-color: #ffb000;
    color: black;
    font-size: 24px;
    font-weight: bold;
    text-align: center;
    width: 200px;
    padding: 30px 0;
    margin: 0 110px;
}
.Counter-button--green {
    background-color: #007f23;
    color: white;
    font-size: 24px;
    font-weight: bold;
    text-align: center;
    width: 200px;
    padding: 30px 0;
    margin: 0 110px;
}