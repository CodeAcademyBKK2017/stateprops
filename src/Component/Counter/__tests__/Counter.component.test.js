import React from 'react';
import Counter from '../Counter.component';
import renderer from 'react-test-renderer';

test('CountStatus Snapshot Test',() => {
  let valueData = 0;
  const tree = renderer.create(<Counter >{valueData}</Counter>);
  expect(tree).toMatchSnapshot();
});
