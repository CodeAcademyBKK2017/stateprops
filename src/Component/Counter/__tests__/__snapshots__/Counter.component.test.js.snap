// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountStatus Snapshot Test 1`] = `
<h1
  style={undefined}
/>
`;
